module/workflow/src/Domain/Entity/Workflow.php

```diff
@@ -77,79 +86,103 @@ public function getCode(): string
     }
 
     /**
-     * @param string $code
+     * @param StatusId $id
      *
      * @return bool
      */
-    public function hasStatus(string $code): bool
+    public function hasStatus(StatusId $id): bool
     {
-        return isset($this->statuses[$code]);
+        return isset($this->statuses[$id->getValue()]);
```

Comment:
Should be ``array_key_exists`` because ``isset`` check it with value and when value is equal NULL, false or 0 it will return false instead true

Reply:
There always be object